server/plugins/schema-plugin-ob-oracle/src/main/java/com/oceanbase/odc/plugin/schema/oboracle/parser/OBOracleGetDBTableByParser.java

```diff
@@ -276,44 +282,64 @@ public List<DBTableIndex> listIndexes() {
         for (DBTableIndex idx : this.indexes) {
             // ob oracle only support btree algorithm.
             idx.setAlgorithm(DBIndexAlgorithm.BTREE);
-            if (primaryKey != null && primaryKey.getName().equals(idx.getName())) {
+            String getIndexDDLSql = "SELECT dbms_metadata.get_ddl('INDEX', '" + idx.getName() + "', '" + schemaName
+                    + "') as DDL from dual";
+            // Get index type、global/local attribute and column names by parse index ddl.
+            try {
+                JdbcOperationsUtil.getJdbcOperations(connection).query(getIndexDDLSql, (rs) -> {
+                    String ddl = rs.getString("DDL");
+                    /**
+                     * Obtaining the DDL of the primary key through dbms_metadata.get_ddl will return the DDL of the
+                     * table, which is a kernel bug.
+                     */
+                    if (com.oceanbase.tools.dbbrowser.util.StringUtils.isNotBlank(ddl)
+                            && ddl.startsWith("CREATE TABLE")) {
```

Comment:
what's this mean? start with 'CREATE TABLE'?

Reply:
In oralce 4.x, Obtaining the DDL of the primary key through dbms_metadata.get_ddl will return the DDL of the table, which is a kernel bug.